### DISCUSSION TITLE
Use browser API or Ruby stdlib?

### DISCUSSION THREAD

**Discussion by yerlaser** [ruby2js/ruby2js#116] [Q&A] [answered]
Hi,
I am doing some basic tests of the tech and so far liking it.
Sometimes it seems to be too good to be true.

I have a question.
If am to fetch json from a backend server, should I use browser api (i.e. fetch) or ruby's own methods?
In the former case, what would I get? Can I somehow convert it to Ruby hash or should I pretend that Ruby (thanks to transpiling) gains JS json support?

**Comment by rubys** [comment, marked as answer]
I use the browser API, and pretend that the JSON returned is an [OpenStruct](https://docs.ruby-lang.org/en/3.0.0/OpenStruct.html).  Each of the following transpire to the same thing:

```ruby
json['field']
json[:field]
json.field
```

**Reply by yerlaser** [reply]
That's very cool. Thanks for the fast reply!
